pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -409,325 +467,553 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
         }
     }
 
+    /**
+     * @brief Apply a matrix with controls to the statevector.
+     *
+     * @param arr Pointer to the statevector.
+     * @param num_qubits Number of qubits.
+     * @param matrix Perfect square matrix in row-major order.
+     * @param controlled_wires Control wires.
+     * @param wires Wires the gate applies to.
+     * @param inverse Indicate whether inverse should be taken.
+     */
     template <class PrecisionT>
-    static void applyIdentity(std::complex<PrecisionT> *arr,
-                              const size_t num_qubits,
-                              const std::vector<size_t> &wires,
-                              [[maybe_unused]] bool inverse) {
-        PL_ASSERT(wires.size() == 1);
-        static_cast<void>(arr);        // No-op
-        static_cast<void>(num_qubits); // No-op
-        static_cast<void>(wires);      // No-op
-    }
+    static void
+    applyNCMultiQubitOp(std::complex<PrecisionT> *arr, std::size_t num_qubits,
+                        const std::complex<PrecisionT> *matrix,
+                        const std::vector<std::size_t> &controlled_wires,
+                        const std::vector<std::size_t> &wires, bool inverse) {
+        constexpr std::size_t one{1};
+        const std::size_t n_contr = controlled_wires.size();
+        const std::size_t n_wires = wires.size();
+        const std::size_t nw_tot = n_contr + n_wires;
+        PL_ASSERT(num_qubits >= nw_tot);
+
+        std::vector<std::size_t> all_wires;
+        all_wires.reserve(nw_tot);
```

Comment:
```suggestion
        std::vector<std::size_t> all_wires(nw_tot);
```

Reply:
I can't do that because `std::vector::insert` will change the size, which should remain `nw_tot`.